spec/hash_dot_spec.rb

```diff
@@ -38,5 +38,17 @@
     end
 
     it_behaves_like "an object", -> { { action: :to_dot } }
+
+    context "with a use_default set" do
```

Comment:
Can we augment the shared behavior spec instead? It will require a little more work than normal, I think, because the current behavior doesn't expect arguments. It will also mean, I think, that we have have to call the shared tests twice for each switch, e.g.:

```
it_behaves_like "an object", -> { Hash.use_dot_syntax = true }
it_behaves_like "an object", -> {
  Hash.use_dot_syntax = true
  Hash.hash_dot_use_default = true
}
```

Reply:
The shared specs have tests that would depend on the args passed.  E.g. the ones that specifically look for `NoMethodError`.  It would seem awkward to keep those there and change the expectation depending on the state of `Hash.hash_dot_use_default`.  Is that what you want or would you prefer those be pulled out to their own context?

Reply:
I think that _is_ what I want. I can't think of another way to make sure that the `use_default: true` case will be tested against the remaining normal hash/object behavior.